### PR TITLE
perf(workflow) filter workflow with the `paths` option

### PR DESCRIPTION
**Pull request by lucasfernog**
**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [ ] Feature
- [ ] New Binding Issue #___
- [ ] Code style update
- [x] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes. Issue #___
- [x] No

**The PR fulfills these requirements:**

- [x] It's submitted to the `dev` branch and _not_ the `latest` branch
- [ ] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix: #xxx[,#xxx]`, where "xxx" is the issue number)
- [ ] A change file is added if any packages will require a version bump due to this PR per [the instructions in the readme](https://github.com/tauri-apps/tauri/blob/dev/.changes/readme.md).

If adding a **new feature**, the PR's description includes:
- [ ] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

**Other information:**

### CODE REVIEW

File: .github/workflows/core-lint-fmt.yml [hunk 1]
```diff
@@ -1,10 +1,11 @@
-name: clippy, fmt and eslint check
+name: eslint check
```
Comment (rajivshah3) [inline .github/workflows/core-lint-fmt.yml]:
```suggestion
name: clippy check
```

Comment (lucasfernog) [reply .github/workflows/core-lint-fmt.yml]: you're fast lol

**Review by rajivshah3 — APPROVED** [review]
Good call on rerunning if the workflow file changes 👍

**Review by jbolda — APPROVED** [review]
Looks good to me!

**Pull request merged**